packages/replayio/src/utils/spawnProcess.ts

```diff
@@ -1,18 +1,23 @@
 import { ChildProcess, spawn, SpawnOptions } from "child_process";
+import { Readable } from "stream";
 import { createDeferred, Deferred } from "./async/createDeferred";
 
+function collectData(readable: Readable | null) {
+  const buffers: Uint8Array[] = [];
+  readable?.on("data", data => buffers.push(data));
+  return buffers;
+}
+
 export function spawnProcess(
   executablePath: string,
   args: string[] = [],
   options: SpawnOptions = {},
-  callbacks: {
-    onError?: (error: Error) => void;
-    onExit?: () => void;
```

Comment:
We were already creating a deferred in `spawnProcess` (but never actually using it). So those 2 felt redundant and I decided to simplify this and lean on the returned `deferred.promise` in the caller